CPAC/connectome/connectivity_matrix.py

```diff
@@ -0,0 +1,274 @@
+    tool : str
+
+    Returns
+    -------
+    str or NotImplemented
+    """
+    tool = tools[tool.lower()]  # make case-insensitive
+    cm_method = methods[tool].get(method, NotImplemented)
+    if cm_method is NotImplemented:
+        logger.warning(NotImplementedError(
+            f'{method} has not yet been implemented for {tool} in C-PAC.'
+        ))
+    return cm_method
+
+
+def compute_correlation(time_series, method):
+    """Function to create a numpy array file [1] containing a
+    correlation matrix for a given time series and method.
+
+    Parameters
+    ----------
+    time_series : str
+        path to time series output
+
+    method : str
+        correlation matrix method. See https://fcp-indi.github.io/docs/nightly/user/tse#configuring-roi-time-series-extraction
+        for options and how to configure.
+
+    References
+    ----------
+    .. [1] The NumPy community (2021). NPY format. https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html#npy-format
+    """  # pylint: disable=line-too-long  # noqa E501
+    desc_regex = re.compile(r'\_desc.*(?=\_)')
+
+    data = np.genfromtxt(time_series).T
+
+    existing_desc = desc_regex.search(time_series)
+    if hasattr(existing_desc, 'group'):
+        matrix_file = time_series.replace(
+            existing_desc.group(),
+            '+'.join([
+                existing_desc.group(),
+                method
+            ])
+        ).replace('_timeseries.1D', '_connectome.npy')
+    else:
+        matrix_file = _connectome_name(time_series, method)
+
+    method = _get_method(method, 'Nilearn')
+    if method is NotImplemented:
+        return NotImplemented
+
+    connectivity_measure = ConnectivityMeasure(kind=method)
+    connectome = connectivity_measure.fit_transform([data])[0]
+
+    np.save(matrix_file, connectome)
+    return matrix_file
+
+
+def compute_connectome_nilearn(parcellation, timeseries, method):
+    """Function to compute a connectome matrix using Nilearn
+
+    Parameters
+    ----------
+    parcelation : Niimg-like object
+        http://nilearn.github.io/manipulating_images/input_output.html#niimg-like-objects
+        Region definitions, as one image of labels.
+
+    timeseries : str
+        path to timeseries image
+
+    method: str
+        'Pearson' or 'Partial'
+
+    Returns
+    -------
+    numpy.ndarray or NotImplemented
+    """
+    output = _connectome_name(timeseries, f'Nilearn{method}')
+    method = _get_method(method, 'Nilearn')
+    if method is NotImplemented:
+        return NotImplemented
+    masker = NiftiLabelsMasker(labels_img=parcellation,
+                               standardize=True,
+                               verbose=True)
+    timeser = masker.fit_transform(timeseries)
+    correlation_measure = ConnectivityMeasure(kind=method)
+    corr_matrix = correlation_measure.fit_transform([timeser])[0]
+    np.fill_diagonal(corr_matrix, 0)
+    np.savetxt(output, corr_matrix)
+    return corr_matrix
```

Comment:
The idea here was to have a Nilearn-only version of `compute_correlation` (I did a bad job naming this function :cold_sweat:)